### PR TITLE
[CDF-21657] Add status filter to workflow executions list

### PR DESCRIPTION
**Pull request by VerstraeteBert**
## Description
Allows filtering data workflow executions by status

## Checklist:
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by silvavelosa — APPROVED** [review]
a test would be nice 👍 .... otherwise LGTM ✔️

File: cognite/client/_api/workflows.py [hunk 1]
```diff
@@ -212,6 +212,7 @@ def list(
         workflow_version_ids: WorkflowVersionIdentifier | MutableSequence[WorkflowVersionIdentifier] | None = None,
         created_time_start: int | None = None,
         created_time_end: int | None = None,
+        statuses: list[Literal["completed", "failed", "running", "terminated", "timed_out"]] | None = None,
```
Comment (doctrino) [inline cognite/client/_api/workflows.py:215]: Can you also allow a single status to be passed in, not just a list? It is a minor thing, but makes the users life a bit easier

Comment (doctrino) [reply cognite/client/_api/workflows.py:215]: Any reason why this list is different than? https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/data_classes/workflows.py#L24

Comment (VerstraeteBert) [reply cognite/client/_api/workflows.py:215]: The statuses there are the statuses for Tasks. The naming is a bit unfortunate. Would be a potential breaking change.

File: cognite/client/data_classes/workflows.py [hunk 2]
```diff
@@ -32,6 +32,8 @@
     "skipped",
 ]
 
+WorkflowStatus: TypeAlias = Literal["completed", "failed", "running", "terminated", "timed_out"]
```
Comment (VerstraeteBert) [inline cognite/client/data_classes/workflows.py:35]: "paused" has been removed from the possible statuses. 
https://github.com/cognitedata/service-contracts/pull/2535

**Review by doctrino — APPROVED** [review]
Nice clarification between Task and Workflow status. In addition, to a user friendly input. Good work

**Pull request merged**